tests/nexus_workflow_test.go

```diff
@@ -2271,7 +2271,7 @@ func (s *NexusWorkflowTestSuite) TestNexusCallbackAfterCallerComplete() {
 
 	run, err := s.SdkClient().ExecuteWorkflow(ctx, client.StartWorkflowOptions{
 		TaskQueue:          taskQueue,
-		WorkflowRunTimeout: 200 * time.Millisecond,
+		WorkflowRunTimeout: 1700 * time.Millisecond,
```

Comment:
Hmm... is this because we truncate the request timeout to the workflow run timeout duration?
This test seems flaky to me, if the CI server is busy, we may not be able to deliver the Nexus StartOperation call.

Let's change this test to:
- The handler workflow waits on a signal for completion
- The caller workflow doesn't use a `WorkflowRunTimeout` at all
- The test waits for the handler workflow to be started and then terminates the caller workflow and signals the handler workflow to unblock